aten/src/ATen/native/native_functions.yaml

```diff
@@ -1479,6 +1479,27 @@
 - func: diff.out(Tensor self, int n=1, int dim=-1, Tensor? prepend=None, Tensor? append=None, *, Tensor(a!) out) -> Tensor(a!)
   variants: function
 
+- func: gradient.scalarint(Tensor self, *, Scalar? spacing=None, int? dim=None, int edge_order=1) -> Tensor[]
+  variants: function
+
+- func: gradient.scalararray(Tensor self, *, Scalar spacing, int[] dim, int edge_order=1) -> Tensor[]
+  variants: function
+
+- func: gradient.array(Tensor self, *, int[] dim, int edge_order=1) -> Tensor[]
+  variants: function
+
+- func: gradient.scalarrayint(Tensor self, *, Scalar[] spacing, int? dim=None, int edge_order=1) -> Tensor[]
+  variants: function
+
+- func: gradient.scalarrayarray(Tensor self, *, Scalar[] spacing, int[] dim, int edge_order=1) -> Tensor[]
+  variants: function
+
+- func: gradient.tensorarrayint(Tensor self, *, Tensor[] spacing, int? dim=None, int edge_order=1) -> Tensor[]
+  variants: function
+
+- func: gradient.tensorarray(Tensor self, *, Tensor[] spacing, int[] dim, int edge_order=1) -> Tensor[]
+  variants: function
```

Comment:
Are you going to add out variants of these functions? If so, consider making it structured https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md

Reply:
BTW, this is an annoyingly large number of overloads; maybe this was in the issue desc

Reply:
By the way, i think the signature ordering we have added in order to fix issue regarding {h/v/d}split methods, causing problem here. As we can see currently CI tests tells our signature matching goes to Tensor[] while we would hope to go for int[]. However after undo'ing the change i have proposed in the previous PR (https://github.com/pytorch/pytorch/pull/57346) this problem no more happens. 

What would you think is a good solution at this point ?
Should we return back to split methods / ordering function, or there is some idea that would be good to apply ?

Reply:
Reverting doesn't sound right. It sounds like your problem is you want `Scalar[]` to be tested before `Tensor[]`: *so say that in the ordering*.

Reply:
Thank you for your suggestion :) and i have updated ordering accordingly. Hope it looks good now :)

Reply:
> Are you going to add out variants of these functions? If so, consider making it structured https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md

Thanks for pointing the link. Currently adding methods are not planned, but will definitely use the help of this link when we are going to add it.